benchmarks/manoptuna.jl

```diff
@@ -0,0 +1,215 @@
+using Manifolds
+using PythonCall
+include("benchmark_comparison.jl")
+
+# This script requires optuna to be available through PythonCall
+# You can install it for example using
+# using CondaPkg
+# ]conda add optuna
+
+optuna = pyimport("optuna")
+
+struct TTsuggest_int
+    suggestions::Dict{String,Int}
+end
+function (s::TTsuggest_int)(name::String, a, b)
+    return s.suggestions[name]
+end
+struct TTsuggest_float
+    suggestions::Dict{String,Float64}
+end
+function (s::TTsuggest_float)(name::String, a, b; log::Bool=false)
+    return s.suggestions[name]
+end
+struct TTsuggest_categorical
+    suggestions::Dict{String,Any}
+end
+function (s::TTsuggest_categorical)(name::String, vals)
+    return s.suggestions[name]
+end
+struct TTreport
+    reported_vals::Vector{Float64}
+end
+function (r::TTreport)(val, i)
+    return push!(r.reported_vals, val)
+end
+struct TTshould_prune end
+(::TTshould_prune)() = Py(false)
+struct TracingTrial
+    suggest_int::TTsuggest_int
+    suggest_float::TTsuggest_float
+    suggest_categorical::TTsuggest_categorical
+    report::TTreport
+    should_prune::TTshould_prune
+end
+
+"""
+    ObjectiveData
+
+
+"""
+mutable struct ObjectiveData{TObj,TGrad}
```

Comment:
This is still undocumented and the functor below _quite_ technical. Could we maybe document that a bit more to also make it understandable for either me or also you in some future time ;) ?

Reply:
I will document it, I'm just not sure what style of describing to use here. I will add a tutorial-style commentary.